### PR TITLE
Update RouteCollection::getPath

### PR DESCRIPTION
**Pull request by WinterSilence**
This version work faster - old code create closure at every calling getPath

### CODE REVIEW

File: src/Http/RouteCollection.php [hunk 1]
```diff
@@ -80,16 +80,19 @@ public function getRouteData()
         return $this->dataGenerator->getData();
     }
 
-    public function getPath($name, $parameters = [])
+    protected function fixPathPart(&$part, $key, array $parameters)
+    {
+        if (is_array($part) && array_key_exists($part[0], $parameters)) {
```
Comment (franzliedke) [inline src/Http/RouteCollection.php]: Are you sure the `array_key_exists()` call is actually necessary?

File: src/Http/RouteCollection.php [hunk 2]
```diff
@@ -80,16 +80,19 @@ public function getRouteData()
...
-        $path = implode('', array_map(function ($part) use ($parameters) {
-            if (is_array($part)) {
-                $part = $parameters[$part[0]];
-            }
-            return $part;
-        }, $parts));
+        array_walk($parts, [$this, 'fixPathPart'], $parameters);
+        $path = implode('', $parts);
 
         $path = '/' . ltrim($path, '/');
```
Comment (franzliedke) [inline src/Http/RouteCollection.php]: You can combine these two lines, I'd say. :)

File: src/Http/RouteCollection.php (hunk 1 above)
Comment (WinterSilence) [inline src/Http/RouteCollection.php]: Validate incoming parameters guarantee correct execution method.

File: src/Http/RouteCollection.php (hunk 2 above)
Comment (WinterSilence) [inline src/Http/RouteCollection.php]: array_walk returns boolean value

Comment (franzliedke) [inline src/Http/RouteCollection.php]: I meant the two below that:

~~~php
$path = '/'.ltrim(implode('', $parts), '/');
~~~

Comment (WinterSilence) [inline src/Http/RouteCollection.php]: ups)) updated

File: src/Http/RouteCollection.php [hunk 3]
```diff
@@ -80,18 +80,18 @@ public function getRouteData()
...
+            $part = $parameters[$part[0]];
+        }
+    }
 
-        $path = '/' . ltrim($path, '/');
+    public function getPath($name, array $parameters = [])
+    {
+        $parts = $this->reverse[$name][0];
+        array_walk($parts, [$this, 'fixPathPart'], $parameters);
+        $path = '/' . ltrim(implode('', $parts), '/');
```
Comment (franzliedke) [inline src/Http/RouteCollection.php]: Just return it directly ;)

Comment (franzliedke) [inline src/Http/RouteCollection.php]: Will do that for you.

**Pull request merged**